<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModbusDialog</class>
 <widget class="QDialog" name="ModbusDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>165</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Modbus</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>false</bool>
  </property>
  <property name="modal">
   <bool>false</bool>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>130</y>
     <width>181</width>
     <height>32</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>11</x>
     <y>17</y>
     <width>381</width>
     <height>95</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QLabel" name="label_13">
      <property name="text">
       <string>地    址：</string>
      </property>
      <property name="buddy">
       <cstring>spinBoxAddr</cstring>
      </property>
     </widget>
    </item>
    <item row="0" column="1" colspan="2">
     <widget class="QSpinBox" name="spinBoxAddr">
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
    </item>
    <item row="0" column="3">
     <widget class="QLabel" name="label_14">
      <property name="text">
       <string>功    能：</string>
      </property>
      <property name="buddy">
       <cstring>comboBoxFun</cstring>
      </property>
     </widget>
    </item>
    <item row="0" column="4" colspan="2">
     <widget class="QComboBox" name="comboBoxFun">
      <item>
       <property name="text">
        <string>3(读保持寄存器)</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>4(读输入寄存器)</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>10(写多保持寄存器)</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>06(写单保持寄存器)</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>0f(写多线圈)</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>05(写单线圈)</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>01(读单线圈)</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>02(读离散量)</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_20">
      <property name="text">
       <string>字节数：</string>
      </property>
      <property name="buddy">
       <cstring>spinBoxLength</cstring>
      </property>
     </widget>
    </item>
    <item row="2" column="3">
     <widget class="QLabel" name="label_17">
      <property name="text">
       <string>写入值：</string>
      </property>
      <property name="buddy">
       <cstring>lineEditValue</cstring>
      </property>
     </widget>
    </item>
    <item row="2" column="4" colspan="2">
     <widget class="QLineEdit" name="lineEditValue">
      <property name="enabled">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QLabel" name="label_16">
      <property name="text">
       <string>寄存器个数：</string>
      </property>
      <property name="buddy">
       <cstring>spinBoxCount</cstring>
      </property>
     </widget>
    </item>
    <item row="1" column="4" colspan="2">
     <widget class="QSpinBox" name="spinBoxCount">
      <property name="maximum">
       <number>65535</number>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_15">
      <property name="text">
       <string>起始寄存器：</string>
      </property>
      <property name="buddy">
       <cstring>spinBoxStart</cstring>
      </property>
     </widget>
    </item>
    <item row="1" column="1" colspan="2">
     <widget class="QSpinBox" name="spinBoxStart">
      <property name="maximum">
       <number>65535</number>
      </property>
     </widget>
    </item>
    <item row="2" column="1" colspan="2">
     <widget class="QSpinBox" name="spinBoxLength">
      <property name="enabled">
       <bool>false</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QComboBox" name="comboBoxPro">
   <property name="geometry">
    <rect>
     <x>60</x>
     <y>120</y>
     <width>131</width>
     <height>31</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>Modbus RTU</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Modbus TCP</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_21">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>125</y>
     <width>51</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>协议：</string>
   </property>
   <property name="buddy">
    <cstring>spinBoxLength</cstring>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ModbusDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ModbusDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
